milestone-3/frontend/src/components/recipePage.tsx

```diff
@@ -0,0 +1,71 @@
+import  { ChangeEvent, useState } from 'react';
+import { Link } from "react-router-dom";
+
+interface Recipe {
+    name: string;
+    image: string;
+    description: string;
+    ingredients: string [];
+    instructions: string [];
+}
+
+function RecipePage(props: Recipe) {
+    
+
+  const [newIngredient, setNewIngredient] = useState('');
+  const [allIngredients, setAllIngredients] = useState(props.ingredients);
+  const [newInstruction, setNewInstruction] = useState('');
+  const [allInstructions, setAllInstructions] = useState(props.instructions);
+
+    return (
+    <div>
+    <h1 className="text">{props.name}</h1>
+    <div className="sub-section">
+        <img className="recipe-images-main" src={props.image} alt="picture of recipe" width="200" height="200"/>
+        <p className="text">
+            {props.description}
+        </p>
+
+    </div>
+    <div className="sub-section">
+        <strong className="text">Ingredients</strong>
+         <ul>
+            {allIngredients.map(ingredient => 
+            <li className="text" key={ingredient}>{ingredient}</li>)}
+        </ul>
+    </div>
+    <input
+        placeholder="enter ingredient here"
+        value={newIngredient} 
+        onChange={(e: ChangeEvent<HTMLInputElement>) => {
+        setNewIngredient(e.target.value);
+    }}
+    />
+     <button onClick={() => setAllIngredients([...allIngredients, newIngredient])}>
```

Comment:
it can be nice for user experience to also reset your newIngredient object to the empty string once you submit so you clear the input field.